packages/backend-function/src/factory.ts

```diff
@@ -226,6 +255,18 @@ class FunctionFactory implements ConstructFactory<AmplifyFunction> {
 
     return this.props.runtime;
   };
+
+  private resolveSchedule = () => {
+    if (!this.props.schedule) {
+      return [];
+    }
+
+    const schedules = Array.isArray(this.props.schedule)
```

Comment:
Why are we converting from **FunctionSchedule | FunctionSchedule[]** to **FunctionSchedule[]** here?

resolveSchedule should only be setting default values if no value is specified, it should not change the accepted types passed into the construct

Reply:
looks like this is duplicated here https://github.com/aws-amplify/amplify-backend/pull/1527/files#diff-e354c6c4617ce06e0e16948f4521eca5f9269445265b80ed16720372d336c886R379.

We should remove this logic from the hydrateDefaults flow

Reply:
Yeah the reason for this was to have less duplication of branching logic to handle users entering either a single schedule or array of schedules. In hindsight now I am thinking of moving the logic to convert to `FunctionSchedule[]` in `convertFunctionSchedulesToRuleSchedules` and having that function able to take in either `FunctionSchedule` or `FunctionSchedule[]`.

Reply:
Done in https://github.com/aws-amplify/amplify-backend/pull/1527/commits/ca2d0bd1eb466fc5752cb4b830375af273372dec